cardano-node-emulator/test/Plutus/Examples/EscrowSpec.hs

```diff
@@ -0,0 +1,493 @@
+
+-- noLockProof :: NoLockedFundsProof EscrowModel
+-- noLockProof = defaultNLFP
+--   { nlfpMainStrategy   = finishingStrategy (const True)
+--   , nlfpWalletStrategy = finishingStrategy . (==) }
+
+-- prop_NoLockedFunds :: Property
+-- prop_NoLockedFunds = checkNoLockedFundsProofWithOptions defInitialDist params noLockProof
+
+-- | Check that you can't redeem after the deadline and not refund before the deadline.
+validityChecks :: ThreatModel ()
+validityChecks = do
+  let startTime = TimeSlot.scSlotZeroTime def
+      params = escrowParams startTime
+      deadline = fromIntegral . TimeSlot.posixTimeToEnclosingSlot def $ escrowDeadline params
+      scriptAddr = Scripts.validatorCardanoAddressAny Params.testnet $ typedValidator params
+  input <- anyInputSuchThat $ (scriptAddr ==) . addressOf
+  rmdr <- (fromData . toPlutusData =<<) <$> getRedeemer input
+  case rmdr of
+    Nothing -> fail "Missing or bad redeemer"
+    Just Impl.Redeem ->
+      shouldNotValidate $
+        changeValidityRange
+          ( TxValidityLowerBound ValidityLowerBoundInBabbageEra deadline
+          , TxValidityNoUpperBound ValidityNoUpperBoundInBabbageEra
+          )
+    Just Impl.Refund ->
+      shouldNotValidate $
+        changeValidityRange
+          ( TxValidityNoLowerBound
+          , TxValidityUpperBound ValidityUpperBoundInBabbageEra (deadline - 1)
+          )
+
+prop_validityChecks :: Actions EscrowModel -> Property
+prop_validityChecks = E.checkThreatModelWithOptions defInitialDist params validityChecks
+
+tests :: TestTree
+tests =
+  testGroup
+    "escrow"
+    -- [ let con = void $ payEp @() @EscrowSchema @EscrowError (escrowParams startTime) in
+    --   checkPredicateOptions options "can pay"
+    --     ( assertDone con (Trace.walletInstanceTag w1) (const True) "escrow pay not done"
+    --     .&&. walletFundsChange w1 (Value.adaValueOf (-10))
+    --     )
+    --     $ do
+    --       hdl <- Trace.activateContractWallet w1 con
+    --       Trace.callEndpoint @"pay-escrow" hdl (Ada.adaValueOf 10)
+    --       void $ Trace.waitNSlots 1
+
+    -- , let con = void $ selectEither (payEp @()
+    --                                        @EscrowSchema
+    --                                        @EscrowError
+    --                                        (escrowParams startTime))
+    --                                 (redeemEp (escrowParams startTime)) in
+    --   checkPredicateOptions options "can redeem"
+    --     ( assertDone con (Trace.walletInstanceTag w3) (const True) "escrow redeem not done"
+    --       .&&. walletFundsChange w1 (Value.adaValueOf (-10))
+    --       .&&. walletFundsChange w2 (Value.adaValueOf 10)
+    --       .&&. walletFundsChange w3 mempty
+    --     )
+    --     redeemTrace
+
+    -- , checkPredicateOptions options "can redeem even if more money than required has been paid in"
+
+    --       -- in this test case we pay in a total of 40 lovelace (10 more than required), for
```

Comment:
Can we uncomment it in the future or should we just delete it? (same question for the other example below).

Reply:
I wanted to have this visible in one commit, so it is visible what we don't support. I will remove it in the next PR.